src/TagDependencyTrait.php

```diff
@@ -161,22 +186,21 @@ public static function loadModel(
      * Invalidate model tags.
      * @return bool
      */
-    public function invalidateTags()
+    public function invalidateTags($event)
     {
-
         /** @var TagDependencyTrait $this */
         \yii\caching\TagDependency::invalidate(
             $this->getTagDependencyCacheComponent(),
             [
                 static::commonTag(),
-                $this->objectTag()
+                $this->objectTag($event->name == ActiveRecord::EVENT_AFTER_UPDATE ? $event->changedAttributes : [])
             ]
         );
 
         if (!empty($this->cacheCompositeTagFields())) {
             \yii\caching\TagDependency::invalidate(
                 $this->getTagDependencyCacheComponent(),
-                $this->objectCompositeTag()
+                $this->objectCompositeTag($event->name == ActiveRecord::EVENT_AFTER_UPDATE ? $event->changedAttributes : [])
```

Comment:
In addition.
1. Please use one code format.
2. Don't use a variable without defining.
3. Use PHP strict mode
4. Test your code